src/features/device-state/routes/state-patch-v2.ts

```diff
@@ -275,6 +275,7 @@ export const statePatchV2: RequestHandler = async (req, res) => {
 						deviceBody = truncateShortTextFields(deviceBody);
 						// If we're updating anyway then ensure the metrics data is included
 						deviceBody = { ...deviceBody, ...metricsBody };
+						deviceBody.cpu_id = deviceBody.cpu_id?.toLowerCase();
```

Comment:
I think you should only do the assignment if the property exists, otherwise you'll be assigning undefined and that could cause funky behavior:
```suggestion
						if (deviceBody.cpu_id != null) {
							deviceBody.cpu_id = deviceBody.cpu_id.toLowerCase();
						}
```
(and the same for v3)